discovery/marathon/marathon.go

```diff
@@ -140,37 +135,28 @@ func NewDiscovery(conf SDConfig, logger log.Logger) (*Discovery, error) {
 		logger = log.NewNopLogger()
 	}
 
-	tls, err := httputil.NewTLSConfig(conf.TLSConfig)
+	client, err := httputil.NewHTTPClientFromConfig(conf.HTTPClientConfig, "marathon_sd")
 	if err != nil {
 		return nil, err
 	}
 
-	token := string(conf.BearerToken)
-	if conf.BearerTokenFile != "" {
-		bf, err := ioutil.ReadFile(conf.BearerTokenFile)
+	// Unfortunately, we have to do this part ourselves because Marathon uses a
```

Comment:
I don't think it's a good idea to override the behaviour of the common HTTP client, bearer token should mean bearer token. If there's a bespoke auth method that some SD uses I think it should have different configuration field names to avoid confusion.

Reply:
Sure, that's reasonable. The [DC/OS documentation](https://docs.mesosphere.com/1.11/security/oss/iam-api/) uses the term "authentication token". So Perhaps `auth_token` / `auth_token_file`?

How should backwards compatibility be handled (because this is currently supported through the standard property names)?

* no backwards compatibility, users will have to rename the configuration properties
* support both the old and new property names and log a warning
* ?

Reply:
> How should backwards compatibility be handled (because this is currently supported through the standard property names)?

Is basic auth supported by Marathon natively? I didn't see it in the docs, so the things to do might be to leave things as-is.

Reply:
Marathon itself only supports basic authentication as per its [documentation here](https://mesosphere.github.io/marathon/docs/ssl-basic-access-authentication.html). Marathon deployed as part of DC/OS only supports token-based authentication (as you noticed in the other documentation).

So arguably, it is the token-based authentication that should be dropped from `marathon_sd`. I wouldn't suggest doing that, since that would break existing functionality and leave no workaround.

Another complication is that the DC/OS documentation says they will eventually support the `Bearer` authorization header. This would also be supported with this PR, but it also means that we cannot easily be backwards compatible and future compatible with a new DC/OS version.

Perhaps this PR is the best way to go forward?

* supports basic auth for marathon itself
* supports token-based authentication for DC/OS
* supports future `Bearer` authentication for DC/OS

Existing DC/OS users will have to rename their configuration properties from `bearer_token` to `auth_token` when deploying the new version. Not doing so will simply fail authenticating...

Reply:
That does seem like the best approach overall, even if it does involve a breaking change.

Reply:
Sounds good.

Is there anything else required to move this forward? Should this log a warning of some sort when `bearer_token` is set? Should I also make the changes to the documentation?

Reply:
I don't see a need for a warning. Docs should be updated in the same PR.

Reply:
I've added 2 commits:

* one that updates the documentation and updates configuration validation accordingly
* another that updates `CHANGELOG.md` to document the breaking change

I'm not 100% sure what the process is around `CHANGELOG.md`, I can drop that commit if you prefer.